### PR TITLE
Add social cards editing

### PR DESCRIPTION
**Pull request by dcsaszar**
Converted and simplified from Scrivito Example App.

I tried to stick with the original as closely as possible.
The styles look slightly different and may need some adjustments or rework.

### Content todo
~~* [ ] Fill in the missing descriptions that cause validation warnings. Could be a follow-up.~~
* [x] Set relevant pages to index: Yes.

### CODE REVIEW

File: src/utils/getMetadata.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  if (ogTitle) meta.push({ property: 'og:title', content: ogTitle })
+
+  return meta
+}
+
+function imageUrlFor(obj: unknown) {
+  if (obj instanceof Obj && obj.objClass() === 'Image' && obj.get('blob')) {
+    return obj.contentUrl()
+  }
+}
```
Comment (apepper) [inline src/utils/getMetadata.ts:77]: FYI: For the `obj instanceof Obj && obj.objClass() === 'Image'` part I would prefer to introduce a `isImage` function similar to:

https://github.com/Scrivito/scrivito-portal-app/blob/2fb056eb94a9a3b07f9b4b0971283adcf2762b25/src/Objs/Homepage/HomepageObjClass.ts#L29-L33

File: src/utils/getMetadata.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  if (description) {
+    meta.push({ name: 'description', content: description })
+  }
+
+  const root = Obj.root()
+  if (root) {
+    const facebookAppId = ensureString(root.get('facebookAppId'))
+    if (facebookAppId) {
+      meta.push({ property: 'fb:app_id', content: facebookAppId })
+    }
```
Comment (apepper) [inline src/utils/getMetadata.ts:27]: `facebookAppId` is not defined on "Homepage". Please add it.

File: src/utils/getMetadata.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    const twitterSite = ensureString(root.get('twitterSite'))
+    if (twitterSite) {
+      meta.push({ name: 'twitter:site', content: twitterSite })
+    }
+  }
```
Comment (apepper) [inline src/utils/getMetadata.ts:33]: Same with `twitterSite`.

File: src/Objs/defaultPageEditingConfig.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    (metaDataDescription: string) => {
+      if (metaDataDescription.length === 0) {
+        return {
+          message: 'The metaDataDescription should be set.',
+          severity: 'warning',
+        }
+      }
+    },
+  ],
```
Comment (apepper) [inline src/Objs/defaultPageEditingConfig.ts:62]: This validation is too restrictive. Please remove it.

File: src/Objs/defaultPageEditingConfig.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+export const defaultPagePropertiesGroups = [
+  {
+    title: 'Metadata',
+    properties: ['metaDataDescription', 'robotsIndex'],
+    key: 'metadata-group',
+  },
+  {
+    title: 'Social cards',
+    component: SocialCardsTab,
+    properties: ['tcCreator', 'tcDescription', 'ogDescription'],
```
Comment (apepper) [inline src/Objs/defaultPageEditingConfig.ts:31]: If I'm not mistaken, this list is not complete. E.g. `tcImage` is missing.

File: src/Objs/defaultPageAttributes.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+export const defaultPageAttributes = {
```
Comment (apepper) [inline src/Objs/defaultPageAttributes.ts:1]: Currently this is only used on `Homepage` and `Page`.

I think the following Obj types would also benefit from this:
* `SubnavigationOverview`
* `ProductOverview`, `ProductCategory` and `Product`

**Review by apepper — CHANGES_REQUESTED** [review]
FEEDBACK ([8286dd2](https://github.com/Scrivito/scrivito-portal-app/pull/461/commits/8286dd266c1ec48bd2aac1319b81d6f3ef1ec1e5))

Nice start 🚀!

File: src/Objs/Homepage/HomepageObjClass.ts [hunk 7]
```diff
@@ -20,6 +21,7 @@ export const Homepage = provideObjClass('Homepage', {
     sitePortalOverviewPage: 'reference',
     siteSearchResultsPage: 'reference',
     siteUserProfilePage: 'reference',
+    twitterSite: 'string',
```
Comment (apepper) [inline src/Objs/Homepage/HomepageObjClass.ts:26]: Please prefix both attributes with `site`, e.g. `siteFacebookAppId` and `siteTwitterSite`.

**Review by apepper — COMMENTED** [review]
FEEDBACK ([71326f7](https://github.com/Scrivito/scrivito-portal-app/pull/461/commits/71326f76a3dca94a69dfcbcd80707639c9ece31a))

Nice 🚀! Getting closer.

Please see my comment and also provide a working copy, where `robotsIndex` is set by most (or all?) pages.

**Comment by dcsaszar** [comment]
> provide a working copy, where `robotsIndex` is set by most (or all?) pages.

All:
https://edit-beta.scrivito.com/d0a154d76edf2a7bd991fc658e700a1d~https://dave-social-cards.scrivito-portal-app.pages.dev/en?_scrivito_workspace_id=p902f870e3545df7&_scrivito_display_mode=view

* [ ] FYI there are now also five pages with a description > 175. I can fix this as soon as I can create more working copies. 😉

**Review by apepper — APPROVED** [review]
LGTM ([c14b5c4](https://github.com/Scrivito/scrivito-portal-app/pull/461/commits/c14b5c41e4e0f806ded6eb704ef58449c81c02fa))

Nice 🚀! Code-wise all good. I'll write separately about the needed code-changes.

**Pull request merged**